compiler/passes/initializerRules.cpp

```diff
@@ -199,9 +201,26 @@ void errorOnFieldsInArgList(FnSymbol* fn) {
 
     for_vector(SymExpr, se, symExprs) {
       if (se->symbol() == fn->_this) {
-        USR_FATAL_CONT(se,
-                       "invalid access of class member in "
-                       "initializer argument list");
+        bool error = true;
+        if (fn->isCopyInit()) {
+          if (CallExpr* call = toCallExpr(se->parentExpr)) {
+            AggregateType* at = toAggregateType(fn->_this->getValType());
+            if (call->isPrimitive(PRIM_TYPEOF)) {
```

Comment:
this.x.type / this.type.x.type ?

Reply:
This fails because "<type-expr>.<var field>" doesn't yet work in any context.